app/services/hyrax/manifest_builder_service.rb

```diff
@@ -0,0 +1,93 @@
+require 'iiif_manifest'
+
+module Hyrax
+  # A class responsible for converting a Hyrax::Work like thing into a IIIF
+  # manifest.
+  #
+  # @see !{.as_json}
+  class ManifestBuilderService
+    # @api public
+    #
+    # @param presenter [Hyrax::WorkShowPresenter] the work presenter from which
+    #        we'll build a manifest.
+    # @param iiif_manifest_factory [Class] a class that initializes with presenter
+    #        object and returns an object that responds to `#to_h`
+    # @param expires_in [Integer] the number of seconds until the cache expires
+    # @param cache_manifest [Boolean] should we perform Rails level caching of
+    #        the generated manifest
+    #
+    # @note While the :presenter may be a Hyrax::WorkShowPresenter it is likely
+    #       defined by Hyrax::WorksControllerBehavior.show_presenter
+    #
+    # @note Why the public class method? It is easier to perform stubs in
```

Comment:
it seems a bit rigid compared with

```ruby
  class_attribute :iiif_manifest_builder
  self.iiif_manifest_builder = Hyrax::ManifestBuilderService.new

  ...
  
  def manifest
    self.class.iiif_manifest_builder.as_json(presenter: presenter)
  end
```

in the above case, if i want to configure my builder on the class level, i have the option to pass arguments at the initializer without changing calls in the business logic. 

e.g. if i want to turn off caching for one controller presently (or in the tests), i'd need to pass `cache_manifest: false` every time i call `#as_json`. with an instance, i can do `MyController.iiif_manifest_builder = Hyrax::ManifestBuilderService.new(cache_manifest: false)` and leave other code untouched. the API footprint for a generic controller is reduced here (at least nominally) since the `#as_json` method now doesn't need to know anything about caching. Liskov compliant subclasses are easier to implement, in this case.